java-checks/src/main/java/org/sonar/java/checks/aws/AwsBuilderMethodFinder.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.aws;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.sonar.java.model.JUtils;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+public abstract class AwsBuilderMethodFinder extends IssuableSubscriptionVisitor {
+
+  private static final String SDK_CLIENT_TYPE = "software.amazon.awssdk.core.SdkClient";
+  private static final String SDK_CLIENT_BUILDER_TYPE = "software.amazon.awssdk.utils.builder.SdkBuilder";
+  protected static final String AWS_CLIENT_BUILDER_TYPE = "software.amazon.awssdk.awscore.client.builder.AwsClientBuilder";
+  private static final MethodMatchers BUILD_METHOD = MethodMatchers.create()
+    .ofSubTypes(SDK_CLIENT_BUILDER_TYPE)
+    .names("build")
+    .addWithoutParametersMatcher()
+    .build();
+
+  protected MethodMatchers getBuildMethod() {
+    return BUILD_METHOD;
+  }
```

Comment:
Wen we use `BUILD_METHOD` directly instead of wrapping it in this method?

Reply:
Indeed, I will remove the method